### PR TITLE
Ignore PyPy's implementation differences in base object class, fix 9225

### PR DESCRIPTION
**Pull request by dansut**
Fix for [#9225](https://twistedmatrix.com/trac/ticket/9225) by causing `twisted.python.reflect.accumulateMethods()` to ignore the base object class, which normally has no methods but under PyPy looks like it does due to implementation differences.

Should fix 3 tests:  
- test_reflect.AccumulateMethodTests.test_baseClass
- test_reflect.AccumulateMethodTests.test_ownClass
- test_reflect.PrefixedMethodsTests.test_onlyObject

### CODE REVIEW

File: src/twisted/python/reflect.py [hunk 1]
```diff
@@ -134,7 +134,8 @@ def accumulateMethods(obj, dict, prefix='', curClass=None):
     if not curClass:
         curClass = obj.__class__
     for base in curClass.__bases__:
-        accumulateMethods(obj, dict, prefix, base)
+        if base is not object:
```
Comment (rodrigc) [inline src/twisted/python/reflect.py]: I don't understand this change.  Can you add a comment here as to why this change is correct?
Does this change work on Cpython 2.7 and Cpython 3.x?

**Review by rodrigc — CHANGES_REQUESTED** [review]

**Review by rodrigc — APPROVED** [review]

File: src/twisted/python/reflect.py (hunk 1 above)
Comment (rodrigc) [reply src/twisted/python/reflect.py]: OK, I did some tests on CPython 2.7 and CPython 3.6 and this change works.
It needs a comment though.

Comment (dansut) [reply src/twisted/python/reflect.py]: Sorry, should have mentioned I had run tests in various versions
